src/Compilers/Core/Portable/Diagnostic/DiagnosticDescriptor.cs

```diff
@@ -199,6 +199,38 @@ public override int GetHashCode()
         }
 
         /// <summary>
+        /// Gets the effective severity of diagnostics created based on this descriptor and the given <see cref="CompilationOptions"/>.
+        /// </summary>
+        /// <param name="compilationOptions">Compilation options</param>
+        public ReportDiagnostic GetEffectiveSeverity(CompilationOptions compilationOptions)
+        {
+            if (compilationOptions == null)
+            {
+                throw new ArgumentNullException(nameof(compilationOptions));
+            }
+
+            var effectiveDiagnostic = compilationOptions.FilterDiagnostic(Diagnostic.Create(this, Location.None));
```

Comment:
The API is meant to give just the effective severity for the descriptor for given compilation options (not compilation, which needs to take into account the source suppressions and pragmas).

Probably the filter functionality can be better refactored to work on descriptor rather than the diagnostic, but I'll leave that work to the compiler team (as it is purely a refactoring). Let me open a new item to track that work. Once implemented, we can avoid creating a dummy no-location diagnostic here.